app/src/renderer/indexed_model.rs

```diff
@@ -0,0 +1,45 @@
+use alloc::vec::Vec;
+use ogc_rs::{print, println};
+use wavefront::{Obj, Vertex};
+
+/**
+ * Our representation of a model.
+ */
+pub struct IndexedModel {
+    pub vertices: Vec<f32>,
+    pub indices: Vec<u16>,
+}
+
+/**
+ * Implementation of the indexed model.
+ */
+impl IndexedModel {
+    /**
+     * Turn a model into its indexed equivalent
+     */
+    pub fn new(object: &Obj) -> IndexedModel {
+        let mut vertex_map: Vec<Vertex> = Vec::new();
+        let mut positions: Vec<f32> = Vec::new();
+        let mut indices: Vec<u16> = Vec::new();
+        for vertex in object.vertices() {
```

Comment:
There might be a way to make the implementation of this loop cleaner. (like by using only iterator functions and no mutable variables) but I do not think the extra time spent writing that is worth it.

Reply:
I'll see if I can improve this when I start work on textured rendering next :)